### DISCUSSION TITLE
dnsLookup cache when using http2

### DISCUSSION THREAD

**Discussion by NicholasBertazzonAga** [sindresorhus/got#1944] [Q&A] [answered]
#### What would you like to discuss?

Hello, I'm currently using your library but I have a problem with dnsLookup. I don't want the result to be cached, and using http2 the custom result I put into dnsLookup is cached. dnsLookup isn't called anymore after the first time for the same url.
Meanwhile, setting http2: false, dnsLookup works as intended and doesn't cache anything.

Maybe the problem is the underlaying http2 nodejs library?
I don't know.

Using latest got 12 beta 4, latest nodejs 16.13

Thanks

**Comment by rrichardson** [comment, marked as answer]
The default for http2 is to leave the connection open.  Are you sure that it is creating a new TCP connection to the same endpoint? Or is it simply re-using the existing connection to send a new request?
